### ISSUE TITLE
When filter is enabled, causes elgato cam link pro and all usb devices to stop working

### ISSUE THREAD

**Issue by kjv13** [exeldro/obs-source-record#36]
When this filter is enabled the following occurs:

- Elgato Cam Link Pro will show no sources
- All usb devices stop working

This happens for about 5 seconds, then for one second everything works, then for another 5 seconds the same behavior repeats.

If you leave it alone, obs will crash. [here](https://pastebin.com/G5Rtb0Xd) is the obs crash log.

This result is bad because it :

- leaves the computer unusable
- it makes it so that the elgato cam link pro sources only work for about a second every five seconds
- causes obs to crash which means even the main recording functionality will not work.

I have tried every encoder including x264. The same behavior exists, I have tried reinstalling the plugin, same behavior.

**Comment by kjv13** [comment]
I have done some more testing and I can verify that this only happens with my elgato cam link pro, and not any of my other sources(e.g. my webcam)

The weird thing is that if my scene only contains my webcam; if I apply the obs-source-record filter to the webcam, it works.

However, if my scene contains BOTH e.g.
-webcam
-elgato cam link pro camera

The stuttering happens when I apply the filter on the cam link pro. The stuttering also happens if I apply it on the webcam. If the elgato cam link pro is in the scene, and if a sourcerecord filter is applied ANYWHERE, the issue starts.

So basically it is an incompatibility between the obs-source-record plugin and the elgato cam link pro.
